packages/react-native/scripts/react_native_pods_utils/script_phases.sh

```diff
@@ -103,8 +103,17 @@ generateArtifacts () {
 moveOutputs () {
     mkdir -p "$RCT_SCRIPT_OUTPUT_DIR"
 
+    # Test if 'cp' supports the '-X' flag (GNU's cp doesn't, but macOS's cp does)
+    touch /tmp/test-cp-X-flag-one /tmp/test-cp-X-flag-two
+    if cp -X /tmp/test-cp-X-flag-one /tmp/test-cp-X-flag-two > /dev/null 2>&1; then
```

Comment:
Thanks for this change! 🙂

Can we test by `[[ "$OSTYPE" == "darwin"* ]]` instead? I appreciate this does direct feature detection, but would prefer if we avoided the file system side effects.

Reply:
Hi @huntie. I agree that a way of feature detection without file systems side effect is desireable.

> Can we test by [[ "$OSTYPE" == "darwin"* ]] instead?

Unfortunately the `OSTYPE` detection is not enough, as it identifies the OS but not the specific implementation of `cp` that's being used. The problem can happen when the OS is Darwin **and** the `cp` command in the `PATH` does not support the `-X` flag (e.g. GNU's `cp`).

The use of temporary files test was chosen since macOS's native `cp` does not provide a `--help` or `--version` command to use instead. It does print its usage on wrong input, but it also emits a non-zero exit code, which is a problem because `script_phases.sh` has `pipefail` set. 

In short, making another test is a bit convoluted and I'm not sure if I have the bandwidth to come up with a working solution for that.

Reply:
I think perhaps we can replace this with the EdenFS detection we use in [Codegen's `build.sh`](https://github.com/facebook/react-native/blob/a4d6be89083598ea4fba8f07699fe905d0c2d163/packages/react-native-codegen/scripts/oss/build.sh#L33) — this way it's more clear why we are applying this flag against a bulk copy.

I'm happy to test this once imported.

```sh
    # Detect if we are in an EdenFS checkout
    if [ -x "$(command -v eden)" ] && eden info >/dev/null 2>&1; then
        CP_FLAGS="-X"
    else
        CP_FLAGS=""
    fi
```